### DISCUSSION TITLE
KISS-over-TCP Port

### DISCUSSION THREAD

**Discussion by leonardinux** [Xastir/Xastir#216] [Q&A] [answered]
KISS TNCs can be accessed over a TCP/IP connection. Xastir miss this port type support. There are many software that can interface KISS modem to TCP ports, such as ser2net or share-tnc or tnc-server (all source are hosted here in GitHub). How can be defined a remote TCP link to a KISS device in Xastir? Simply read a TCP address that read/write KISS frames... but I cannot find a solution into Interfaces menu...
Any trick? have I missed a previous thread about it?

73 de Leo IZ5FSA

**Comment by tvrusso** [comment]
There is no capability built into Xastir to connect to a KISS-over-TCP port.  One could clearly be added, but is is unlikely that anyone has the cycles to do it anytime soon.

That said, there may be simple ways to get the job done, but you'll have to involve an intermediary.

One could cobble together a kludge using "socat" to connect a KISS-over-TCP TNC to a pseudo TTY, then connect Xastir to the PTY as if it were a serial KISS TNC.  See http://www.dest-unreach.org/socat/doc/socat.html for the man page.  Simply have socat talk to the KISS tnc's TCP/IP port for one of its interfaces and a PTY for the other, then tell Xastir to use the slave side of the PTY as if it were a serial KISS TNC.

**Comment by leonardinux** [comment]
Thanks!!!
Googling a bit I've found this solution:
$ socat PTY,link=/tmp/tnc TCP:192.168.1.10:8001 &
This command "pipe" /tmp/tnc to a network TCP address.
In Xastir you can use a Serial KISS TNC on /tmp/tnc device and all can work!
I think I'll work around this feature to create a "script" that can be
started/stopped to render it fully automated by Xastir itself.
Stay tuned!

[image: immagine.png]

73 de Leo IZ5FSA

Il giorno mar 24 ago 2021 alle ore 03:42 Tom Russo ***@***.***>
ha scritto:

> There is no capability built into Xastir to connect to a KISS-over-TCP
> port. One could clearly be added, but is is unlikely that anyone has the
> cycles to do it anytime soon.
>
> That said, there may be simple ways to get the job done, but you'll have
> to involve an intermediary.
>
> One could cobble together a kludge using "socat" to connect a
> KISS-over-TCP TNC to a pseudo TTY, then connect Xastir to the PTY as if it
> were a serial KISS TNC. See
> http://www.dest-unreach.org/socat/doc/socat.html for the man page. Simply
> have socat talk to the KISS tnc's TCP/IP port for one of its interfaces and
> a PTY for the other, then tell Xastir to use the slave side of the PTY as
> if it were a serial KISS TNC.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Xastir/Xastir/issues/179#issuecomment-904251746>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABNGA44G36AHDYFRDJFAACTT6L2JDANCNFSM5CVRFSVA>
> .
>


-- 
*--*
*leonardinux*

**Comment by leonardinux** [comment, marked as answer]
Hi OM,
I've create this script to run my Xastir and it seems to be ok:
`#!/bin/bash
socat PTY,link=/tmp/tnc TCP:192.168.1.10:8001 &
sleep 1
xastir
sleep 1
kill -SIGTERM $(pidof socat)
#
`
73 de Leo IZ5FSA

**Comment by tvrusso** [comment]
Since this issue is of continued interest and is less visible as a closed bug report, I am converting it to a discussion so it is more easily located.

**Reply by tvrusso** [reply]
I have also added information about use of KISS-over-TCP TNCs to the new Github wiki, provided a link to this discussion there, and quoted Leo's script in that documentation.
